src/dos/drive_local.cpp

```diff
@@ -748,17 +748,12 @@ bool localFile::UpdateDateTimeFromHost()
 	if (fstat(file, &temp_stat) == -1)
 		return true; // use defaults
 
-	const tm *ltime = localtime(&temp_stat.st_mtime);
-	if (!ltime)
+	struct tm ltime;
+	if (!safe_localtime(&temp_stat.st_mtime, &ltime))
 		return true; // use defaults
 
-	time = DOS_PackTime(static_cast<uint16_t>(ltime->tm_hour),
-	                    static_cast<uint16_t>(ltime->tm_min),
-	                    static_cast<uint16_t>(ltime->tm_sec));
-
-	date = DOS_PackDate(static_cast<uint16_t>(ltime->tm_year + 1900),
-	                    static_cast<uint16_t>(ltime->tm_mon + 1),
-	                    static_cast<uint16_t>(ltime->tm_mday));
+	time = DOS_PackTime(ltime);
+	date = DOS_PackDate(ltime);
```

Comment:
Suggest making the `struct tm` variable name match whatever is we decide to use per above comment